src/minio/managers/user_manager.py

```diff
@@ -490,3 +516,53 @@ async def _delete_user_home_directory(self, username: str) -> None:
                 await self.client.delete_object(bucket_name, obj_key)
 
             logger.info(f"Deleted {len(objects)} objects from {user_prefix}")
+
+    async def _delete_user_system_directory(self, username: str) -> None:
+        """Delete user system directories and all contents (like Spark job logs)."""
+
+        # Get only user-scoped system paths to avoid deleting global resources
+        system_paths = self._get_user_system_paths(username, user_scoped_only=True)
+
+        # Delete directories for each system bucket and prefix
+        for bucket_name, prefixes in system_paths.items():
+            if not await self.client.bucket_exists(bucket_name):
+                continue  # Skip if bucket doesn't exist
+
+            for prefix in prefixes:
+                # List all objects in user system directory
+                objects = await self.client.list_objects(bucket_name, prefix)
+
+                # Delete objects
+                for obj_key in objects:
+                    await self.client.delete_object(bucket_name, obj_key)
+
+                logger.info(
+                    f"Deleted {len(objects)} objects from s3a://{bucket_name}/{prefix}/"
+                )
+
+    def _get_user_system_paths(
+        self, username: str, user_scoped_only: bool = False
+    ) -> Dict[str, List[str]]:
+        """Get system resource paths for a user using the global configuration.
+
+        Args:
+            username: The username to get paths for
+            user_scoped_only: If True, only returns user-scoped paths (ending with /{username}).
+                              If False, returns all system paths for directory creation.
+        """
+        user_paths = defaultdict(list)
+
+        for _, resource_config in SYSTEM_RESOURCE_CONFIG.items():
+            bucket = resource_config["bucket"]
+            base_prefix = resource_config["base_prefix"]
+            user_scoped = resource_config.get("user_scoped", True)
+
+            # Skip global resources if only user-scoped paths are requested
+            if user_scoped_only and not user_scoped:
+                continue
+
+            # Generate path based on whether resource is user-scoped
+            path = f"{base_prefix}/{username}" if user_scoped else base_prefix
+            user_paths[bucket].append(path)
+
+        return user_paths
```

Comment:
```suggestion
        return dict(user_paths)
```

Return a std dict vs defaultdict